crates/utils/src/error.rs

```diff
@@ -147,7 +150,10 @@ pub enum LemmyErrorType {
   CommunityHasNoFollowers,
   PostScheduleTimeMustBeInFuture,
   TooManyScheduledPosts,
-  FederationError(Option<FederationError>),
+  FederationError {
+    #[cfg_attr(feature = "full", ts(optional))]
+    error: Option<FederationError>,
+  },
```

Comment:
Had to make these named structs, because tuple structs don't support the derives.